packages/react-widgets/src/widgets/GeocoderWidget.js

```diff
@@ -53,12 +53,17 @@ const SearchIcon = (args) => (
  * @param  {object} props
  * @param  {Object} [props.className] - Material-UI withStyle class for styling
  * @param  {Function} [props.onError] - Function to handle error messages from the widget.
+ * @param  {Object=} [props.credentials] - Credentials to use, otherwise widgets checks redus state.
+ * @param  {string=} props.method - method, either `sql` or `lds`, see geocodeStreetPoint
+ * @param  {Boolean=} [props.zoomControl] - Optional, default true. Whether control should zoom map on result.
+ *
+ * @see geocodeStreetPoint
  */
 function GeocoderWidget(props) {
   const inputRef = useRef();
   const oauthCredentials = useSelector(selectOAuthCredentials);
   const globalCredentials = useSelector((state) => state.carto.credentials);
-  const credentials = oauthCredentials || globalCredentials;
+  const credentials = props.credentials || oauthCredentials || globalCredentials;
```

Comment:
Why do we need to add credentials and method to the props? I guessed that the implementation of this would be just check the globalCredentials apiVersion and if it's V3 it would use lds and if it isn't use sql. Idk if I'm missing something. Idk why we have oauthCredentials here ?¿ The global credentials is created using those credentials.

Reply:
Actually, this is just proposition, not particularly tied to this approach.

I've later found that i can push those credentials through redux too, if that's problematic i can remove it and rely only on `carto.credentials` ...

Reply:
After brief discussion, we agreed with @Clebal that this component should use Redux `carto.credentials`.
We can set it using `setCredentials` action from `react-redux`.
It's cleaner and better aligned with rest of ecosystem.

Reply:
Allright then, we'll see how / where that is set from cloud-native side